### PR TITLE
Redo: Add profiling of already running tasks via SIGINFO/SIGUSR1

### PR DESCRIPTION
**Pull request by IanButterworth**
Redo of https://github.com/JuliaLang/julia/pull/43179 (which has been reverted)
I overlooked buildkite build hangs.

It turns out the precompile script revised during review was bad but for some reason it didn't error on the buildbots, and may have caused the hangs on buildkite. TBC.

### CODE REVIEW

**Comment by IanButterworth** [comment]
This appears to have fixed the issue, but CI is behind

File: stdlib/Profile/src/Profile.jl [hunk 1]
```diff
@@ -34,6 +34,50 @@ macro profile(ex)
...
+        end
+    end
+end
+
+# An internal function called to show the report after an information request (SIGINFO or SIGUSR1).
+function _peek_report()
+    iob = IOBuffer()
+    ioc = IOContext(IOContext(iob, stdout), :displaysize=>displaysize(stdout))
+    print(ioc, groupby = [:thread, :task])
+    Base.print(stdout, String(resize!(iob.data, iob.size)))
```
Comment (vtjnash) [inline stdlib/Profile/src/Profile.jl:57]: I am not sure where you found the privacy-violating `resize!` accessor, but we have `take!` for this:
```
    Base.print(stdout, String(take!(iob.data)))
```

File: src/signal-handling.c [hunk 2]
```diff
@@ -129,6 +129,35 @@ static size_t jl_safe_read_mem(const volatile char *ptr, char *out, size_t len)
     return i;
 }
 
+static double profile_autostop_time = -1.0;
+static double profile_peek_duration = 1.0; // seconds
+
+double jl_get_profile_peek_duration(void) {
```
Comment (vtjnash) [inline src/signal-handling.c:135]: style nit: functions start with `{` on the next line
```suggestion
double jl_get_profile_peek_duration(void)
{
```

File: src/signal-handling.c [hunk 3]
```diff
@@ -129,6 +129,35 @@ static size_t jl_safe_read_mem(const volatile char *ptr, char *out, size_t len)
...
+    return profile_peek_duration;
+}
+void jl_set_profile_peek_duration(double t) {
+    profile_peek_duration = t;
+    return;
```
Comment (vtjnash) [inline src/signal-handling.c:140]: style nit: return is not needed
```suggestion
```

File: stdlib/Profile/src/Profile.jl (hunk 1 above)
Comment (IanButterworth) [reply stdlib/Profile/src/Profile.jl:57]: Great question. I usually do `take!` I'll try to find where I copied that from.

Comment (IanButterworth) [reply stdlib/Profile/src/Profile.jl:57]: I can't find where I got it from.. weird. 

Applying suggestions https://github.com/JuliaLang/julia/pull/44199

**Pull request merged**